### PR TITLE
Remove powermock

### PR DESCRIPTION
**Pull request by timja**
Due to not respecting module boundaries powermock does not work out of the box in Java 16+. On Java 11+ it had a poor user experience with annotations, guess work and googling required to make it work at all.

Personally I block new powermock tests and if they break I normally remove them / replace them with something else, and I don't seem to be the only one, e.g. @MarkEWaite removed them from the git plugin, https://github.com/jenkinsci/git-plugin/pull/1139#discussion_r711577465.

@jglick suggested removing powermock from the plugin pom as an encouragement for maintainers when they upgrade to remove their tests or look for different options

Powermock has not had a merged PR in nearly 1 year at this point.

Here's a sample of some concerning issues:

https://github.com/powermock/powermock/issues/992#issuecomment-917442102
https://github.com/powermock/powermock/issues/1053
https://github.com/powermock/powermock/issues/969

Mailing list is very in-active:
https://groups.google.com/g/powermock

Options for people coming here:

1. Use a `JenkinsRule` or a `RealJenkinsRule` and `TestExtension` where necessary which will provide more realistic behaviour but slower tests
2. Use dependency injection with the facade pattern, example: https://github.com/jenkinsci/github-checks-plugin/search?q=facade cc @uhafner 
3. Use mockito or mockito-inline (https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#39), mockito can now mock static methods and final classes and it is actively maintained. Example PRs: https://github.com/jenkinsci/jenkins/pull/5736, https://github.com/jenkinsci/git-plugin/pull/1139 
4. Just remove the tests, often you have better things to do than fight with mocking frameworks that may or may not be adding a lot of value
5. Stay with powermock, you'll need to just add an explicit version yourself, in the future if you wish to test against java 16+ you can configure maven-surefire plugin to add the required `add-opens`, example https://github.com/powermock/powermock/issues/969#issuecomment-914205216

Depending on what people think we could bump this to 5.x as well, maybe a time to get something like https://github.com/jenkinsci/plugin-pom/pull/133 revived (but with less compatibility and hackery)?

### CODE REVIEW

**Comment by jglick** [comment]
see #441 also

**Comment by jglick** [comment]
> we could bump this to 5.x as well

If you like. I do not personally find the numbering all that helpful. Sometimes there are changes that will break some child POMs and you need to do a bit of work to upgrade; that is what release notes are for.

File: src/it/powermock/src/test/java/test/XTest.java [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.powermock.api.mockito.PowerMockito;
-import org.powermock.core.classloader.annotations.PowerMockIgnore;
-import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
-
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(Jenkins.class)
-@PowerMockIgnore({"com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*"})
```
Comment (jglick) [inline src/it/powermock/src/test/java/test/XTest.java:16]: Ah yes this mess, best left in the rear-view mirror.

Do we care about adding a Mockito IT? To #441 I guess, so prove that the inline static stuff works?

Comment (timja) [reply src/it/powermock/src/test/java/test/XTest.java:16]: I don't think so =/.

Seems a bit weird to add something like that to me. Powermock was more of a special case even so I'm not sure having it here was the best, it wasn't a magic fix to read it from the parent pom and it worked across java versions, you needed the above mess all over the place.

**Review by basil — APPROVED** [review]

**Comment by timja** [comment]
> And don't get me started on the enormous pita trying to upgrade the Mockito tests to the newer version of the framework 😓 If I ever get forced to do that upgrade I don't know what to do :/

This doesn't force you to upgrade 😄, you'll just need to add an explicit dependency in your plugin.

**Review by jetersen — APPROVED** [review]

**Review by uhafner — APPROVED** [review]

**Pull request merged**